### DISCUSSION TITLE
Redirection of thumbnail viewer's output to a file

### DISCUSSION THREAD

**Discussion by gardockt** [pystardust/ytfzf#583] [Ideas]
Currently, when a thumbnail viewer outputs something, such as an error, it writes to the display, messing up the interface. This is especially noticeable with asynchronous thumbnail loading, as trying to display a partially downloaded thumbnail prints an error (at least when using `ueberzug`). Another, more consistent way to reproduce this is to use `imv` and press **p** while its window is focused - a newline character is then printed. While this can be solved by redirecting viewer's output to a file, doing so also has some negative consequences.

Advantages:
+ display is not messed up by viewer's output
+ when debugging is needed, viewer's output can be read fully, with no characters getting overwritten (possibly low importance, as most significant error details can be read anyway)

Disadvantages:
- no direct indication of viewer throwing errors, which may result in viewer-related issue reports not containing potentially valuable information
- getting to read viewer's output is less convenient

(obviously, any other programs that `ytfzf` forks should be considered as well - thumbnail viewers may not be the only case)

What do you think about this change?

**Comment by Euro20179** [comment]
Maybe there could be some kind of variable like `thumbnail_debug_log=/dev/null` then it can be changed to `/dev/stderr` or something else if output is wanted.
It could have a companion option `--thumbnail-log` to set the variable.

**Reply by gardockt** [reply]
So I understand that having to ask for additional data in case of issues (which is probably the most important disadvantage) is not a problem - I can create a pull request in a few hours.

**Reply by Euro20179** [reply]
Asking for more data is not really  a problem, as  I can simply ask them to run `ytfzf --thumbnail-log=/dev/stdout`.

Also, there is no need for a pull request as I have already implemented this :)
